File: src/org/jointheleague/memory/controller/GameController.java
package org.jointheleague.memory.controller;

import org.jointheleague.memory.model.GameModel;

import java.awt.event.WindowAdapter;
import java.awt.event.WindowEvent;

public class GameController extends WindowAdapter {

    private final GameModel model;

    public GameController(GameModel model) {
        this.model = model;
    }

    public void buttonClicked(int i) {
        model.select(i);
    }

    @Override
    public void windowClosing(WindowEvent e) {
        System.exit(0);
    }

    public void onGameOver(boolean playAgain) {
        if (playAgain) {
            model.reset();
        } else {
            System.exit(0);
        }
    }
}

File: src/org/jointheleague/memory/controller/GameController.kt
package org.jointheleague.memory.controller

import org.jointheleague.memory.model.GameModel

import java.awt.event.WindowAdapter
import java.awt.event.WindowEvent
import kotlin.system.exitProcess

class GameController(private val model: GameModel) : WindowAdapter() {

    fun buttonClicked(i: Int) {
        model.select(i)
    }

    override fun windowClosing(e: WindowEvent?) {
        exitProcess(status = 0)
    }

    fun onGameOver(playAgain: Boolean) {
        if (playAgain) {
            model.reset()
        } else {
            exitProcess(status = 0)
        }
    }
}